src/widget/whotcuebutton.h

```diff
@@ -2,12 +2,41 @@
 
 #include <QString>
 
+#include "track/trackid.h"
 #include "util/parented_ptr.h"
 #include "widget/wcuemenupopup.h"
 #include "widget/wpushbutton.h"
 
 class WHotcueButton : public WPushButton {
     Q_OBJECT
+
+    struct HotcueDragInfo {
+        HotcueDragInfo(TrackId id, int cue)
+                : trackId(id),
+                  hotcue(cue){};
+
+        HotcueDragInfo(QByteArray& bytes) {
+            QDataStream stream(&bytes, QIODevice::ReadOnly);
+            QVariant idVar;
+            stream >> idVar >> hotcue;
+            trackId = TrackId(idVar);
+        };
+
+        const QByteArray toByteArray() {
```

Comment:
no need to return a value as const.
```suggestion
        QByteArray toByteArray() {
```